### ISSUE TITLE
Nightly features being used

### ISSUE THREAD

**Issue by ion-elgreco** [viz-rs/path-tree#43]
All our builds in delta-rs are failing because nightly features are being used in the latest release of path-tree, see https://github.com/delta-io/delta-rs/actions/runs/12536673265/job/34959932714?pr=3090#step:4:1702

```rust
error[E0658]: use of unstable library feature 'is_none_or'
   --> C:\Users\runneradmin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\path-tree-0.8.2\src\parser.rs:112:65
    |
112 | ...                   let suffix = self.cursor.peek().is_none_or(|(_, c)| *c == '/');
    |                                                       ^^^^^^^^^^
    |
    = note: see issue #126383 <https://github.com/rust-lang/rust/issues/126383> for more information
```

@fundon

**Comment by fundon** [comment]
`is_none_or` landed in `1.82`.

https://blog.rust-lang.org/2024/10/17/Rust-1.82.0.html
https://doc.rust-lang.org/nightly/std/option/enum.Option.html#method.is_none_or

Can you upgrade MSRV?

**Comment by fundon** [comment]
`v0.8.2` has been yanked.
